fairroot/base/sink/FairRootFileSink.cxx

```diff
@@ -202,9 +202,7 @@ void FairRootFileSink::TruncateBranchNames(TBranch* b, TString ffn)
 
 void FairRootFileSink::Close()
 {
-    if (fRootFile) {
-        fRootFile->Close();
-    }
+    fRootFile->Close();
```

Comment:
Please keep the old `if (fRootFile)`. Or can we guarantee that this pointer is always non-nullptr?

Reply:
I also thought about this. The fRootFile is owned by the sink object anyway and it can't be nullptr when this method is called. Of course, someone could deliberately delete the pointer obtained from the getter. But in this crazy case, we should just let program crash instead of letting the program keep running and making the debugging difficult.

Reply:
I would still prefer the "save" version, please.

Reply:
Well, there is no concept of "safety" if the program is already used in an extremely unsafe way. The best way to deal with this case, IMO, is let the program crash as soon as possible such that the bug can be more easily found.

I also said something wrong in the last comment. If a user delete the pointer, `if(fRootFile)` won't help here because `fRootFile` is a non-nullptr pointer pointing to an invalid memory. And I don't think there is any way that a user can set `fRootFile` to be a nullptr. As long as the source class lives, `fRootFile` can never be nullptr.